contrib/cirrus/runner.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# This script is intended to be executed by automation or humans
+# under a hack/get_ci_vm.sh context.  Use under any other circumstances
+# is unlikely to function.
+
+set -e
+
+if [[ -r "/etc/automation_environment" ]]; then
+    source /etc/automation_environment
+    source $AUTOMATION_LIB_PATH/common_lib.sh
+else
+    (
+    echo "WARNING: It does not appear that containers/automation was installed."
+    echo "         Functionality of most of ${BASH_SOURCE[0]} will be negatively"
+    echo "         impacted."
+    ) > /dev/stderr
+fi
+
+OS_RELEASE_ID="$(source /etc/os-release; echo $ID)"
+# GCE image-name compatible string representation of distribution _major_ version
+OS_RELEASE_VER="$(source /etc/os-release; echo $VERSION_ID | tr -d '.')"
+# Combined to ease some usage
+OS_REL_VER="${OS_RELEASE_ID}-${OS_RELEASE_VER}"
+
+export "PATH=$PATH:$GOPATH/bin"
+
+podmanmake() {
+    req_env_vars GOPATH SKOPEO_PATH SKOPEO_CI_CONTAINER_FQIN
+    warn "Accumulated technical-debt requires execution inside a --privileged container.  This is very likely hiding bugs!"
+    showrun podman run -it --rm --privileged \
+        -e GOPATH=$GOPATH \
+        -v $GOPATH:$GOPATH:Z \
+        -w $SKOPEO_PATH \
+        $SKOPEO_CI_CONTAINER_FQIN \
+            make "$@"
+}
+
+_run_setup() {
+    if [[ "$OS_RELEASE_ID" == "fedora" ]]; then
+        # This is required as part of the standard Fedora VM setup
+        growpart /dev/sda 1
+        resize2fs /dev/sda1
+
+        # VM's come with the distro. skopeo pre-installed
+        dnf erase -y skopeo
+    else
+        die "Unknown/unsupported distro. $OS_REL_VER"
+    fi
+}
+
+_run_vendor() {
+    podmanmake vendor BUILDTAGS="$BUILDTAGS"
+}
+
+_run_build() {
+    podmanmake bin/skopeo BUILDTAGS="$BUILDTAGS"
+}
+
+_run_validate() {
+    podmanmake validate-local BUILDTAGS="$BUILDTAGS"
+}
+
+_run_unit() {
+    podmanmake test-integration-local BUILDTAGS="$BUILDTAGS"
+}
```

Comment:
_In principle_ these might not need to run in a container, if the top level has all the relevant Go tools in the right versions. Doing it in a container doesn’t hurt, of course.

(`test-integration` more or less needs a container because it starts several registry servers, and `test-system` starts registry servers _in another container_ IIRC.

Reply:
> Doing it in a container doesn’t hurt, of course.

Actually it does.  Unless it's been fixed, there was at least one test that absolutely refused to run correctly outside of a `--privileged` container.  This was missed (for possibly years) precisely *because* the tests only ever ran in a container.

My (strong) preference is that tests run 100% on the host, and then only secondarily run inside a container (if specifically required for the test-subject).  Compromises were made here (and a big-fat 'TODO' added) for the sake of getting off of travis faster.

Reply:
I'll open a PR to drop the container stuffs, let's see what breaks (it's been quite a while since I've tried).

Reply:
Ref: https://github.com/containers/skopeo/pull/1334

Reply:
> Actually it does. Unless it's been fixed, there was at least one test that absolutely refused to run correctly outside of a `--privileged` container.

That should happen for `test-integration`, but not `test-unit`. (That typo identified elsewhere might well have played a part.)

The integration tests don’t care about a _container_ as such, it’s just that they are somewhat invasive (run multiple servers, and the like). If a host opts in by `SKOPEO_CONTAINER_TESTS=1` / `container_magic`, running them directly on the host is fine with me (as long as the host’s environment is kept similar to `Dockerfile`, with low maintenance effort).

Reply:
>  host’s environment is kept similar to Dockerfile, with low maintenance effort

Yep, we're on the same page here.  This is what I was playing with in https://github.com/containers/skopeo/pull/1334 using a script to do the shared setup stuff, then calling the script from both the Dockerfile and Cirrus-CI.  The logistics and setup seem to be working, but the tests are breaking (differently from what I remember before).  In any case, we can continue that design/discussion over in that PR.